Prac_06/car.py

```diff
@@ -0,0 +1,44 @@
+"""
+CP1404/CP5632 Practical
+Car class example
+"""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, vehicle_name, fuel=0):
+        """Initialise a Car instance.
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.name = vehicle_name
+        self.fuel = fuel
+        self.odometer = 0
+
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
+    def drive(self, distance):
+        """Drive the car a given distance.
+        Drive given distance if car has enough fuel
+        or drive until fuel runs out return the distance actually driven.
+        """
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+
+        self.odometer += distance
+        return distance
+
+    def __str__(self):
+        """using f string formatting"""
+        return f'{self.name}, fuel={str(self.fuel)}, km={str(self.odometer)}'

```

Comment:
What examples have we shown you that use str() inside an f-string? (None... it's not needed.)